File: 백준/Gold/2230. 수 고르기/수 고르기.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int N = sc.nextInt();
		int M = sc.nextInt();
		
		int[] arr = new int[N];
		int min = Integer.MAX_VALUE;
		
		for(int i=0; i<N; i++) {
			arr[i] = sc.nextInt();
		}
		
		Arrays.sort(arr);
		
		int start = 0;
		int end = 1;
		
		while(start <= end && end < arr.length) {
			if(arr[end] - arr[start] >= M) {
				min = Math.min(min, arr[end] - arr[start]);
				start++;
			}else {
				end++;
			}
		}
		System.out.println(min);
	}
}

File: 백준/Gold/2230. 수 고르기/수 고르기.kt
fun main(){
    var (N, M) = readln().split(" ").map { it.toInt() }

    var arr = MutableList(N){0}

    for(i in 0 until N){
        arr[i] = readln().toInt()
    }

    var sortArr = arr.sorted()

    var start = 0
    var end = 1

    var min = Integer.MAX_VALUE

    while(start <= end && end < N){
        var dis = sortArr[end] - sortArr[start]

        if(dis >= M){
            min = minOf(min, dis)
            start++
        }else{
            end++
        }
    }

    println(min)

}